core/domain/suggestion_services.py

```diff
@@ -2588,45 +2567,157 @@ def update_question_review_stats(
+                last_contribution_date=_get_contribution_date(
+                    suggestion.last_updated)
             )
-        else:
-            question_review_stat = (
-                _create_question_review_stats_from_model(
-                    question_review_stat_model))
-
-            question_review_stat.reviewed_questions_count += 1
-            question_review_stat.accepted_questions_count += 1 * int(
-                is_accepted)
-            (
-                question_review_stat
-                .accepted_questions_with_reviewer_edits_count
-            ) += 1 + int(suggestion.edited_by_reviewer)
-            question_review_stat.last_contribution_date = (
-                    datetime.datetime.strptime(
-                        str(
-                            suggestion.last_updated.utcnow().date()
-                            .isoformat()),
-                            '%Y-%m-%d'
-                        ).date())
-
-            _update_question_review_stats_models([question_review_stat])
+            continue
+
+        question_review_stat = (
+            _create_question_review_stats_from_model(
+                question_review_stat_model))
+
+        get_incremental_question_review_stats(
+            question_review_stat, suggestion.last_updated,
+            suggestion_is_accepted,
+            suggestion.edited_by_reviewer)
+
+        _update_question_review_stats_models([question_review_stat])
 
     update_question_contribution_stats_at_review(
         suggestion, suggestion.author_id, target_id)
+
+
+def get_incremental_translation_contribution_stats_at_review(
+    translation_contribution_stat: (
+        suggestion_registry.TranslationContributionStats),
+    content_word_count: int,
+    suggestion_is_accepted: bool,
+    edited_by_reviewer: bool
+) -> None:
+    """Updates TranslationContributionStats object.
+
+    Args:
+        translation_contribution_stat: TranslationContributionStats. The stats
+            object to update.
+        content_word_count: int. The number of words in the translation.
+        suggestion_is_accepted: bool. A flag that indicates whether the
+            suggestion is accepted.
+        edited_by_reviewer: bool. A flag that indicates whether the suggestion
+            is edited by the reviewer.
+    """
+    (
+        translation_contribution_stat
+        .accepted_translations_count
+    ) += int(suggestion_is_accepted)
+    (
+        translation_contribution_stat
+        .accepted_translations_without_reviewer_edits_count
+    ) += int(edited_by_reviewer)
+    translation_contribution_stat.accepted_translation_word_count += (
+        content_word_count) * int(suggestion_is_accepted)
+    translation_contribution_stat.rejected_translations_count += int(
+        not suggestion_is_accepted)
+    translation_contribution_stat.rejected_translation_word_count += (
+        content_word_count * int(not suggestion_is_accepted))
```

Comment:
Same notes above re (0 if suggestion_is_accepted) etc.

or you can do something like:

```
 translation_contribution_stat.rejected_translation_word_count += (
     0 if suggestion_is_accepted else content_word_count)
```
which seems cleaner. Similarly elsewhere.

Reply:
Done